java/src/jmri/InstanceManager.java

```diff
@@ -63,12 +63,14 @@
  * @author Bob Jacobsen Copyright (C) 2001, 2008, 2013, 2016
  * @author Matthew Harris copyright (c) 2009
  */
-public class InstanceManager {
+public final class InstanceManager {
 
-    protected static final HashMap<Class<?>, ArrayList<Object>> managerLists = new HashMap<>();
-    private static final InstanceInitializer initializer = new jmri.managers.DefaultInstanceInitializer();
+    // the default instance of the InstanceManager
+    private static volatile InstanceManager defaultInstanceManager = null;
```

Comment:
> having a class initializer also create an instance of the class being initialized

Haven't heard of that as a question. Will do some research, thanks. I had always heard it the other way: static initialization is done by the classloader, hence atomic, so you don't have to worry about synchronization of the Singleton `get`. Perhaps that's no longer an issue with modern JVMs.